sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java

```diff
@@ -573,4 +574,18 @@ public TableClientBuilder tableName(String tableName) {
 
         return this;
     }
+
+    /**
+     * Sets a flag indicating if tenant discovery should be enabled when authenticating with the Storage Table Service.
```

Comment:
We should also provide guidance here in the javadocs on when setting this to `true` is desirable and what is the reason for this to be disabled by default (basically explaining @christothes comment above).